modules/capi-yaml-machine-template-vsphere.adoc

```diff
@@ -0,0 +1,60 @@
+// Module included in the following assemblies:
+//
+// * machine_management/cluster_api_machine_management/cluster_api_provider_configurations/cluster-api-config-options-vsphere.adoc
+
+:_mod-docs-content-type: REFERENCE
+[id="capi-yaml-machine-template-vsphere_{context}"]
+= Sample YAML for a Cluster API machine template resource on {vmw-full}
+
+The machine template resource is provider-specific and defines the basic properties of the machines that a compute machine set creates.
+The compute machine set references this template when creating machines.
+
+#Fields scraped from https://github.com/openshift/cluster-capi-operator/commit/e97432fad394d560b527423b30316e674c20508f#diff-777f6511149ce401729e6065a02ca6e8a89bfc21b0b543fecdb40dd021d227deR188-R204[here]. Values in `spec.template.spec` are taken from the MAPI machine set docs, need validation.#
+
+[source,yaml]
+----
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: VSphereMachineTemplate # <1>
+metadata:
+  name: <template_name> # <2>
+  namespace: openshift-cluster-api
+spec:
+  template:
+    spec: # <3>
+      template: <vm_template_name> # <4>
+      server: <vcenter_server_ip> # <5>
+      diskGiB: 128
+      cloneMode: linkedClone # <6>
```

Comment:
@miyadav need to update [automation script](https://github.com/openshift/openshift-tests-private/blob/master/test/extended/testdata/clusterinfrastructure/capi/machinetemplate-vsphere.yaml#L16) to cover cloneMode

Reply:
yes made a PR. thanks

Reply:
To add more info - 
for mapi machines we get warning like below for linkedClone
`Warning: linkedClone clone mode is set. DiskGiB parameter will be ignored, disk size from template will be used.`
No warning for fullClone.

Reply:
Added similar info to the docs, thanks!